holoviews/element/raster.py

```diff
@@ -530,21 +540,35 @@ class ImageStack(Image):
     _vdim_reductions = {1: Image}
 
     def __init__(self, data, kdims=None, vdims=None, **params):
-        if isinstance(data, np.ndarray) and data.ndim == 3:
-            x = np.arange(data.shape[0])
-            y = np.arange(data.shape[1])
-            data = (x, y, *(data[:, :, n] for n in range(data.shape[2])))
+        _kdims = kdims or self.kdims
+        if isinstance(data, list) and len(data):
+            x = np.arange(data[0].shape[1])
+            y = np.arange(data[0].shape[0])
+            data_tuple = (x, y, *data)
+        elif isinstance(data, dict):
```

Comment:
Technically I think a dictionary containing both key dimension coordinate arrays and value dimension arrays should be accepted here.

Reply:
e.g. something like: `{'x': np.array([0, 0.5, 1]), 'y': np.array([0, 1]), 'z1': np.random.rand(2, 3), 'z2': np.random.rand(2, 3)}`

And you shouldn't have to do any explicit processing to make this work, the grid interface should already support this.

Reply:
Most of the logic in this if statement is to support this case: `{'z1': np.random.rand(2, 3), 'z2': np.random.rand(2, 3)}`, e.g., no kdims coordinates.

Reply:
Yes, but it breaks the case I'm talking about no?

Reply:
It shouldn't. I first try to look up the kdims value in the dictionary, and if those do not exist, I create them. 

![image](https://github.com/holoviz/holoviews/assets/19758978/4e85f3b2-2d6e-4fc7-ae51-a0b83642c6cc)

Maybe I'm just not understanding your question correctly.

Reply:
Nope, you are understanding correctly I just didn't read your code correctly :)